include/deal.II/particles/particle_accessor.h

```diff
@@ -320,9 +320,9 @@ namespace Particles
 
     /**
      * A pointer to the container that stores the particles. Obviously,
-     * this accessor is invalidated if the container changes.
+     * this accessor becomes invalid if the container changes.
      */
-    particle_container *particles;
+    typename particle_container::iterator particles_on_cell;
```

Comment:
This could in principle be much larger than just a pointer. But I guess this is also the most easily portable if we decide to change the data structure again.